src/components/MainNav/MainNav.js

```diff
@@ -83,6 +83,13 @@ class MainNav extends Component {
     this._unsubscribe = this.store.subscribe(() => {
       const { history, location } = this.props;
       const module = this.curModule;
+      const state = this.store.getState();
+
+      // If user has timed out, force them to log in again.
+      if (state.okapi && state.okapi.authFailure
+        && filter(state.okapi.authFailure, { type: 'error', code: 'user.timeout' }).length) {
```

Comment:
@ryandberger would `find` be more appropriate here? You wouldn't need to use `length`.